pkg/config/app/kuma-cp/config.go

```diff
@@ -32,17 +32,31 @@ var _ config.Config = &Config{}
 
 var _ config.Config = &Defaults{}
 
+const (
+	ModeNone DefaultsMode = "none"
+	ModeDemo DefaultsMode = "demo"
+)
+
+type DefaultsMode string
+
 type Defaults struct {
 	config.BaseConfig
 
+	// Mode for defining default resources. Possible values: "none", "demo".
+	// None - no default resources will be created. This is best suited for IaC and production use cases where you want to precisely control what's created.
+	// Demo - default resources will be created. This is best suited for demoing and testing use cases where you want to quickly get started.
+	Mode DefaultsMode `json:"mode" envconfig:"kuma_defaults_mode"`
 	// If true, it skips creating the default Mesh
+	// Deprecated: use mode instead.
```

Comment:
1. There is a way to add deprecation warnings in the logs which we should probably leverage here.
2. be mindful of mink and CD and more generally backward compat
3. in k8s should we auto default to none and have the default stuff be part of the helm chart?

Reply:
1. In config files it looks the same (word, "Deprecated" followed by ":"):

https://github.com/kumahq/kuma/blob/1f6f11bded0b389da6b911d35405cf3f7f59434f/pkg/config/plugins/runtime/k8s/config.go#L137-L139

There is also a warning in Validate() function:

https://github.com/kumahq/kuma/blob/1f6f11bded0b389da6b911d35405cf3f7f59434f/pkg/config/plugins/runtime/k8s/config.go#L448-L450

So I assume it's about adding it in `Validate()` function.

---

will respond to rest of the points in a later comment